src/datamodel/builder/Payload.cpp

```diff
@@ -10,3 +10,98 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "up-cpp/datamodel/builder/Payload.h"
+namespace uprotocol::datamodel::builder {
+
+// Byte vector constructor
+Payload::Payload(const std::vector<uint8_t>& value_bytes,
+                 const v1::UPayloadFormat format) {
+	if (!UPayloadFormat_IsValid(format)) {
+		throw std::out_of_range("Invalid Byte vector payload format");
+	}
+	std::string serialized(value_bytes.begin(), value_bytes.end());
+	payload_ = std::make_tuple(std::move(serialized), format);
+}
+
+// String constructor
+Payload::Payload(const std::string& value, const v1::UPayloadFormat format) {
+	if (!UPayloadFormat_IsValid(format)) {
+		throw std::out_of_range("Invalid String payload format");
+	}
+	payload_ = std::make_tuple(std::move(value), format);
+}
+
+// Move string constructor
+Payload::Payload(std::string&& value, const v1::UPayloadFormat format) {
+	if (!UPayloadFormat_IsValid(format)) {
+		throw std::out_of_range("Invalid RValue String payload format");
+	}
+	payload_ = std::make_tuple(std::move(value), format);
+}
+
+// Move Serialized constructor
+Payload::Payload(Serialized&& serialized) {
+	if (!UPayloadFormat_IsValid(std::get<PayloadType::Format>(serialized))) {
+		throw std::out_of_range("Invalid RValue Serialized payload format");
+	}
+	payload_ = std::move(serialized);
+}
+
+// Move constructor
+Payload::Payload(Payload&& other) noexcept
+    : payload_(std::move(other.payload_)) {}
+
+// Copy constructor
+Payload::Payload(const Payload& other) : payload_(other.payload_) {}
+
+// Move assignment operator
+Payload& Payload::operator=(Payload&& other) noexcept {
+	payload_ = std::move(other.payload_);
+	return *this;
+}
+
+// Copy assignment operator
+Payload& Payload::operator=(const Payload& other) {
+	payload_ = other.payload_;
+	return *this;
+}
```

Comment:
`moved_` also needs to be initialized / copied in these constructors and operators. In the case of the move versions, `other.moved_` should probably also be set to `true`.